qark/plugins/manifest/custom_permissions.py

```diff
@@ -27,53 +29,13 @@ def run(self, file_object):
         for permission in permission_sections:
             try:
                 if permission.attributes["android:protectionLevel"].value in ("signature", "signatureOrSystem"):
-                    min_sdk = get_min_sdk(manifest_xml)
-                    if min_sdk < 21:
-                        self.issues.append(Vulnerability(category=self.category, severity=self.severity,
-                                                         issue_name=self.issue_name, description=self.description,
-                                                         file_object=file_object))
+                    if extras.get("minimum_sdk", get_min_sdk(manifest_xml)) < 21:
+                        self.issues.add(Vulnerability(category=self.category, severity=self.severity,
+                                                      issue_name=self.issue_name, description=self.description,
+                                                      file_object=manifest_file))
 
             except KeyError:
                 continue
 
 
-def get_min_sdk(manifest_xml):
-    """
-    Given the manifest as a `minidom.parse`'d object, try to get the minimum SDK the manifest specifies.
-
-    :param manifest_xml: object after parsing the XML
-    :return: int of the version if it exists, else 1 (the default)
-    """
-    # TODO: try to get SDK from gradle file
-    try:
-        sdk_section = manifest_xml.getElementsByTagName("uses-sdk")[0]
-    except IndexError:
-        log.debug("Unable to get uses-sdk section")
-        return 1
-
-    try:
-        return int(sdk_section.attributes["android:minSdkVersion"].value)
-    except (KeyError, AttributeError):
-        log.debug("Unable to get minSdkVersion from manifest")
-        return 1
-
-
-def get_target_sdk(manifest_xml):
-    """
-    Given the manifest as a `minidom.parse`'d object, try to get the target SDK the manifest specifies.
-
-    :param manifest_xml: object after parsing the XML
-    :return: int of the version if it exists, else 1 (the default)
-    """
-    # TODO: try to get SDK from gradle file
-    try:
-        sdk_section = manifest_xml.getElementsByTagName("uses-sdk")[0]
-    except IndexError:
-        log.debug("Unable to get uses-sdk section")
-        return 1
-
-    try:
-        return int(sdk_section.attributes["android:targetSdkVersion"].value)
-    except (KeyError, AttributeError):
-        log.debug("Unable to get targetSdkVersion from manifest")
-        return 1
\ No newline at end of file
+plugin = CustomPermissions()
```

Comment:
Is this meant to be here? It looks like a global?

Reply:
PluginBase lets us import things as a module, but each plugin will have a different class name. That means that to instantiate it, we need to know the class name. We won't know how to instantiate them with different names, so each plugin module creates an instance of itself so that we can just do 

```python
plugin = plugin_source.load_plugin(plugin_name).plugin
```

To know what plugin to run.

Reply:
Makes sense